starport/pkg/xgenny/sourcemodification.go

```diff
@@ -0,0 +1,58 @@
+package xgenny
+
+// SourceModification describes modified and created files in the source code after a run
+type SourceModification struct {
+	modified map[string]struct{}
+	created  map[string]struct{}
+}
+
+func NewSourceModification() SourceModification {
+	return SourceModification{
+		make(map[string]struct{}),
+		make(map[string]struct{}),
+	}
+}
+
+// ModifiedFiles returns the modified files of the source modification
+func (sm SourceModification) ModifiedFiles() (modifiedFiles []string) {
+	for modified := range sm.modified {
+		modifiedFiles = append(modifiedFiles, modified)
+	}
+	return
+}
+
+// CreatedFiles returns the created files of the source modification
+func (sm SourceModification) CreatedFiles() (createdFiles []string) {
+	for created := range sm.created {
+		createdFiles = append(createdFiles, created)
+	}
+	return
+}
+
+// AppendModifiedFile appends modified files in the source modification that are not already documented
+func (sm *SourceModification) AppendModifiedFiles(modifiedFiles ...string) {
```

Comment:
```suggestion
func (sm *SourceModification) appendModifiedFiles(modifiedFiles ...string) {
```

?

Reply:
Those are actually part of the encapsulated interface, this also allows us to use a separate package for the tests